app/src/main/java/com/example/rickandmortyapp/ui/character/CharacterRepository.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.rickandmortyapp.ui.character
```

Comment:
The CharacterRepository is receiving the RickAndMortyApi already created through the `ApiModule` logic. That means that Hilt will deliver to the CharacterRepository the real instance of the Api that can be consumed.

Instead of using callbacks, we should take advantage of the coroutine's capabilities to do everything in a simple way that feels sequential but still non-blocking and async.